src/config.rs

```diff
@@ -0,0 +1,59 @@
+use anyhow::{Context, Result};
+use hickory_client::client::SyncClient;
+use hickory_client::rr::rdata::tsig::TsigAlgorithm;
+use hickory_client::rr::Name;
+use hickory_client::udp::UdpClientConnection;
+use hickory_proto::rr::dnssec::tsig::TSigner;
+use serde::{Deserialize, Deserializer};
+use serde_with::{base64::Base64, serde_as, DisplayFromStr, MapPreventDuplicates};
+use std::collections::HashMap;
+use std::str::FromStr;
+
+#[serde_as]
+#[derive(Deserialize, Debug)]
+pub(crate) struct Config {
+    pub(crate) resolver: std::net::SocketAddr,
```

Comment:
As discussed in the meeting, this should be an `Option`, which falls back to using https://docs.rs/hickory-resolver/latest/hickory_resolver/type.TokioAsyncResolver.html#method.tokio_from_system_conf

Reply:
Actually hickory-resolver does not allow for txt queries as far as I can tell, and the trait bounds do not seem compatible with hickory-client that easily, it's probably easier to implement reading the resolv.conf myself.